src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Services/SyntaxFacts/ISyntaxFacts.cs

```diff
@@ -81,6 +81,7 @@ internal partial interface ISyntaxFacts
         /// preprocessor directive.  For example `if` or `pragma`.
         /// </summary>
         bool IsPreprocessorKeyword(SyntaxToken token);
+        bool IsPreProcessorDirectiveContext(SyntaxTree syntaxTree, int position, CancellationToken cancellationToken);
```

Comment:
moved this from ISemanticFacts.   This is not a semantic fact.  it's a syntax fact.

Reply:
also, the inconsistency between Preprocessor and PreProcessor bugs me.  but i'm not changing that now.